src/utils/aepp-wallet-communication/connection/index.ts

```diff
@@ -23,63 +23,44 @@
  * import WalletConnection from '@aeternity/aepp-sdk/es/utils/aepp-wallet-communication/connection'
  */
 
-import { required } from '@stamp/required'
-
 /**
  * Basic Wallet Connection
  *
  * This stamp include interface for wallet connection functionality.
  * Attempting to create instances from the Stamp without overwriting all
  * abstract methods using composition will result in an exception.
- * @function
  * @alias module:@aeternity/aepp-sdk/es/utils/aepp-wallet-communication/connection
- * @rtype Stamp
  * @param {Object} [options={}] - Initializer object
  * @return {Object} WalletConnection instance
  */
-export default required({
-  methods: {
-    connect: required,
-    disconnect: required,
-    sendMessage: required,
-    isConnected: required
-  }
-})
-
-/**
- * Connect
- * @function connect
- * @instance
- * @abstract
- * @rtype (onMessage: Function) => void
- * @param {Function} onMessage - Message handler
- * @return {void}
- */
-
-/**
- * Disconnect
- * @function disconnect
- * @instance
- * @abstract
- * @rtype () => void
- * @return {void}
- */
-
-/**
- * Send message
- * @function sendMessage
- * @instance
- * @abstract
- * @rtype (msg: Object) => void
- * @param {Object} msg - Message
- * @return {void}
- */
-
-/**
- * Check if connected
- * @function isConnected
- * @instance
- * @abstract
- * @rtype () => Boolean
- * @return {Boolean} Is connected
- */
+export default interface WalletConnection{
+  /**
+   * Connect
+   * @instance
+   * @abstract
+   * @param onMessage - Message handler
+   * @param onDisconnect - trigger when runtime connection in closed
+   */
+  connect: (onMessage: Function, onDisconnect?: Function) => void
```

Comment:
`Function` is too general, proposing to specify amount of arguments and `: void`?